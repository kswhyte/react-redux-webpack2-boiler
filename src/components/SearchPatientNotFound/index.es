import Component from './searchPatientNotFound';
export default Component;
