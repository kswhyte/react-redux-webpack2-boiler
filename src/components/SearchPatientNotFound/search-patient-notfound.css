@import "imports/_helpers.css";
@import "imports/_layout.css";
@import "imports/_form-elements.css";
@import "imports/_typography.css";

.search-patient-notfound-wrapper {
  & .patient-notfound-msg {
    margin-top: 10px;
  }
  /*& .new-patient-icon {
    background-image: image-set(
      url('~/images/new-patient-icon/-e-new_patient_icon.png') 1x,
      url('~/images/new-patient-icon/-e-new_patient_icon@2x.png') 2x)
    );
  }*/
}
