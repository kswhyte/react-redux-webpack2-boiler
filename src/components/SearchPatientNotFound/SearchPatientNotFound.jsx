import React from 'react';

import './search-patient-notfound.css';

const SearchPatientNotFound = () => {
  return (
    <div className="search-patient-notfound-wrapper">
      <form
        onSubmit={e => {
          e.preventDefault();
          // TODO: potentially pull all info from PatientSearch and populate new form for creating new patient
          // const userFirstName = escape(document.querySelector('#first-name').value);
          // const userLastName = escape(document.querySelector('#last-name').value);
          // props.validationError('');
          // const userData = {
          // firstName,
          // lastName
          // };
          // props.enterNewPatient(userData);
        }}
      >
        <div className="row sm-spacer">
          <div className="col-xs-8 no-margin">
            <h3 className="patient-notfound-msg">This patient is not found in hg smartaccess.</h3>
          </div>
          <div className="col-xs-4 no-margin">
            <div className="new-patient-icon" />
            <button className="primary-btn" type="submit">
              {/* <i className="fa fa-plus-circle" aria-hidden="true" />
              <i className="fa fa-user" aria-hidden="true" /> */}
              Enter New Patient
            </button>
          </div>
        </div>
      </form>
    </div>
  );
};

SearchPatientNotFound.propTypes = {};

export default SearchPatientNotFound;
