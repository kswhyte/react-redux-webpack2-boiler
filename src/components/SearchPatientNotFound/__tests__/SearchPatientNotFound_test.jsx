import React from 'react';
// import { storiesOf, action, linkTo } from '@kadira/storybook';
import SearchPatientNotFound from '../SearchPatientNotFound';

// import { specs, describe, it } from 'storybook-addon-specifications';

import {storiesOf, describe, it, specs
} from "../../../../.storybook/facade";

import { shallow, mount } from 'enzyme';
import expect from 'expect';


const stories = storiesOf('SearchPatientNotFound', module);

stories.add('Search Patient Story', () => {
  const searchPatientNotFoundStory = (
    <SearchPatientNotFound/>);

  specs(() => describe('When a Patient is Not Found', () => {
    describe('Patient NotFound elements and content', () => {
      it('Should render the SearchPatientNotFound component without crashing', () => {
        let output = shallow(searchPatientNotFoundStory);
        expect(output.find('.search-patient-notfound-wrapper').length).toEqual(1);
      });

      it('Should have a header element that says "This Patient is not found in hg smartaccess"', () => {
        let output = mount(searchPatientNotFoundStory);
        expect(output.find('h3').text()).toContain('This Patient is not found in hg smartaccess.');
      });

      it('Should have one call-to-action Create New Patient button', () => {
        let output = mount(searchPatientNotFoundStory);
        expect(output.find('button').length).toEqual(1);
      });
    });
  }));
  return searchPatientNotFoundStory;
});
