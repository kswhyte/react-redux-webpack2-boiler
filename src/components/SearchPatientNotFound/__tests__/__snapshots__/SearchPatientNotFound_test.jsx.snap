// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Patient Story 1`] = `
<div
  className="search-patient-notfound-wrapper"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="row sm-spacer"
    >
      <div
        className="col-xs-8 no-margin"
      >
        <h3
          className="patient-notfound-msg"
        >
          This patient is not found in hg smartaccess.
        </h3>
      </div>
      <div
        className="col-xs-4 no-margin"
      >
        <button
          className="primary-btn"
          type="submit"
        >
          Enter New Patient
        </button>
      </div>
    </div>
  </form>
</div>
`;
